### PR TITLE
Fix for issue1742

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request addresses issue #1742 by introducing an escape argument which, when set to True, will guarantee that all strings are valid HTML.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Entrez/Parser.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
 from xml.parsers import expat
 from io import BytesIO
 import xml.etree.ElementTree as ET
+from xml.sax.saxutils import escape
```
Comment (peterjc) [inline Bio/Entrez/Parser.py]: You don't use this escape module in this file, do you? i.e. Can it be removed.

(This also avoids the potential confusion with an ``__init__`` argument also called ``escape``)

**Comment by peterjc** [comment]
Test structure failure,

```
======================================================================
ERROR: test_custom_directory (test_Entrez.CustomDirectoryTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_Entrez.py", line 168, in test_custom_directory
    handler = Parser.DataHandler(validate=False)
TypeError: __init__() takes exactly 3 arguments (2 given)
======================================================================
ERROR: test_pubmed_html_escaping (test_Entrez_parser.EFetchTest)
Test parsing XML returned by EFetch, PubMed database with HTML tags and HTML escape characters.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_Entrez_parser.py", line 3662, in test_pubmed_html_escaping
    record = Entrez.read(handle, escape=True)
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Entrez/__init__.py", line 463, in read
    record = handler.read(handle)
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Entrez/Parser.py", line 364, in read
    self.parser.ParseFile(handle)
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Entrez/Parser.py", line 752, in externalEntityRefHandler
    parser.ParseFile(handle)
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Entrez/Parser.py", line 742, in externalEntityRefHandler
    handle = _urlopen(url)
  File "run_tests.py", line 265, in dummy_urlopen
    raise RuntimeError("Internal test suite error, attempting to use internet despite --offline setting")
RuntimeError: Internal test suite error, attempting to use internet despite --offline setting
----------------------------------------------------------------------
```

Currently ``Tests/test_Entrez_parser.py`` is an offline test, but ``externalEntityRefHandler`` is trying to fetch something - perhaps a DTD file?

Looking at this now, perhaps this the ``dummy_urlopen`` message (added in #1643) should include the URL?

```python
raise RuntimeError("Internal test suite error, attempting to use internet despite --offline setting")
```

File: Bio/Entrez/Parser.py (hunk 1 above)
Comment (mdehoon) [reply Bio/Entrez/Parser.py]: escape is a function.

Comment (peterjc) [reply Bio/Entrez/Parser.py]: I see it now,

```python
    def characterDataHandlerEscape(self, content):
        content = escape(content)
        self.consumer.consume(content)
```

Sorry, false alarm.

**Comment by peterjc** [comment]
Adding the DTD files fixed the offline test, merged.

**Pull request merged**